### PR TITLE
Feature/add recurring

### PR DESCRIPTION
**Pull request by Nainterceptor**
Checklist

- [x] I've added a mention in CHANGELOG.md

### CODE REVIEW

**Comment by Nainterceptor** [comment]
[Récurrence planning](https://trello.com/c/6GUy6xfD/264-r%C3%A9currence-planning)

File: components/form/recurring-option.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+export default {
+  components: {
+    ecField,
+    ecSwitch,
+    ecSwitchOptions,
+  },
+  props: {
+    value: {
+      type: Object,
+      required: () => true,
```
Comment (Scttpr) [inline components/form/recurring-option.vue]: Isn't it possible to juste use `required: true` here ?

File: components/form/recurring-option.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      default: () => ({ frequency: null, enabled: false }),
+    },
+    frequency: {
+      type: Object,
+      default: () => null,
+    },
+  },
+  data() {
+    return {
+      newValue: this.value,
```
Comment (Scttpr) [inline components/form/recurring-option.vue]: Here I would have deleted this to add a condition in computed setter :
```
if (!this.newValue) { this.newValue = this.value }
```

File: components/form/recurring-option.vue (hunk 1 above)
Comment (Nainterceptor) [reply components/form/recurring-option.vue]: W00t ? I was drunk.

File: components/form/switch-options.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    </button>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    value: {
+      type: String,
+      default: null,
```
Comment (Scttpr) [inline components/form/switch-options.vue]: Here is a different syntax compared to the previous component, is it on purpose ?

File: components/form/recurring-option.vue (hunk 2 above)
Comment (Nainterceptor) [reply components/form/recurring-option.vue]: It's more respectful for the vue lifecycle, and the way used by buefy

File: components/form/switch-options.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    options: {
+      type: Object,
+      default: () => ({}),
```
Comment (Scttpr) [inline components/form/switch-options.vue]: Why don't use null here ?

File: components/form/switch-options.vue [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    },
+  },
+  data() {
+    return {
+      newValue: this.value,
```
Comment (Scttpr) [inline components/form/switch-options.vue]: Same comment as before

File: components/form/switch.vue [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+<script>
+export default {
+  props: {
+    value: {
+      type: Boolean,
+      default: () => false,
+    },
+  },
+  data() {
+    return {
```
Comment (Scttpr) [inline components/form/switch.vue]: Same as before

File: components/form/switch-options.vue (hunk 4 above)
Comment (Nainterceptor) [reply components/form/switch-options.vue]: With a null, template will fail (null is not iterable)

**Pull request merged**